src/machine/state/mod.rs

```diff
@@ -16,6 +17,49 @@ pub struct State {
 
 impl State {
     pub fn new() -> Self {
```

Comment:
I suggest we use the builder pattern, as specified in the machine interface at the moment.